test/test_dists.jl

```diff
@@ -1,245 +1,332 @@
 # Unit tests for Distances
 
-@testset "individual metrics" begin
+macro test_metricity(_dist, _x, _y, _z)
+    quote
+        dist = $(esc(_dist))
+        x = $(esc(_x))
+        y = $(esc(_y))
+        z = $(esc(_z))
+        dxy = evaluate(dist, x, y)
+        dxz = evaluate(dist, x, z)
+        dyz = evaluate(dist, y, z)
+        if isa(dist, PreMetric)
+            # Unfortunately small non-zero numbers (~10^-16) are appearing
+            # in our tests due to accumulating floating point rounding errors.
+            # We either need to allow small errors in our tests or change the
+            # way we do accumulations...
+            @test evaluate(dist, x, x) + one(eltype(x)) ≈ one(eltype(x))
```

Comment:
Also, if you want, this while thing could just be changed to a function instead of a macro.